File: app/src/main/java/kr/kro/awesometic/plankhelper/data/source/PlankLogsDataSource.java
package kr.kro.awesometic.plankhelper.data.source;

import android.support.annotation.NonNull;

import java.util.List;

import kr.kro.awesometic.plankhelper.data.LapTime;
import kr.kro.awesometic.plankhelper.data.PlankLog;

/**
 * Created by Awesometic on 2017-04-16.
 */

public interface PlankLogsDataSource {

    interface LoadPlankLogsCallback {

        void onPlankLogsLoaded(List<PlankLog> plankLogs);

        void onDataNotAvailable();
    }

    interface GetPlankLogCallback {

        void onPlankLogLoaded(PlankLog plankLog);

        void onDataNotAvailable();
    }

    interface SavePlankLogCallback {

        void onSavePlankLog(boolean isSuccess);
    }

    void getPlankLogs(@NonNull LoadPlankLogsCallback callback);

    void getPlankLog(@NonNull String plankLogId, @NonNull GetPlankLogCallback callback);

    void savePlankLog(@NonNull PlankLog plankLog, @NonNull SavePlankLogCallback callback);

    void deletePlankLog(@NonNull String plankLogId);

    void deleteAllPlankLogs();


    // 한 번에 하나의 PlankLog에 대한 모든 LapTime을 불러옴
    interface LoadLapTimesCallback {

        void onLapTimesLoaded(List<LapTime> lapTimes);

        void onDataNotAvailable();
    }

    void getLapTimes(@NonNull String plankLogId, @NonNull LoadLapTimesCallback callback);

    void saveLapTimes(@NonNull String plankLogId, @NonNull List<LapTime> lapTimes);

    void deleteLapTimes(@NonNull String lapTimeId);

    // 해당 PlankLog의 모든 LapTime을 지움
    void deleteAllLapTimes(@NonNull String plankLogId);
}

File: app/src/main/java/kr/kro/awesometic/plankhelper/data/source/PlankLogsDataSource.kt
package kr.kro.awesometic.plankhelper.data.source
/*
import kr.kro.awesometic.plankhelper.data.LapTime
import kr.kro.awesometic.plankhelper.data.PlankLog


interface PlankLogsDataSource {

    interface LoadPlankLogsCallback {

        fun onPlankLogsLoaded(plankLogs: List<PlankLog>)

        fun onDataNotAvailable()
    }

    interface GetPlankLogCallback {

        fun onPlankLogLoaded(plankLog: PlankLog)

        fun onDataNotAvailable()
    }

    interface SavePlankLogCallback {

        fun onSavePlankLog(isSuccess: Boolean)
    }

    fun getPlankLogs(callback: LoadPlankLogsCallback)

    fun getPlankLog(plankLogId: String, callback: GetPlankLogCallback)

    fun savePlankLog(plankLog: PlankLog, callback: SavePlankLogCallback)

    fun deletePlankLog(plankLogId: String)

    fun deleteAllPlankLogs()


    // 한 번에 하나의 PlankLog에 대한 모든 LapTime을 불러옴
    interface LoadLapTimesCallback {

        fun onLapTimesLoaded(lapTimes: List<LapTime>)

        fun onDataNotAvailable()
    }

    fun getLapTimes(plankLogId: String, callback: LoadLapTimesCallback)

    fun saveLapTimes(plankLogId: String, lapTimes: List<LapTime>)

    fun deleteLapTimes(lapTimeId: String)

    // 해당 PlankLog의 모든 LapTime을 지움
    fun deleteAllLapTimes(plankLogId: String)
}
*/